frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -617,6 +617,7 @@ class CallInfo {
                          const ResolutionResultByPostorderID& byPostorder,
                          bool raiseErrors = true,
                          std::vector<const uast::AstNode*>* actualAsts=nullptr,
+                         ID* moduleScopeId=nullptr,
```

Comment:
This needs a comment about what it is, especially since it is set by the function, unlike the other arguments. Also, should it be a `Scope*` ? Lastly, I'm wondering why this `const Scope*` is not added as a field to CallInfo (that is normally `nullptr`). IMO that would be intuitive and make this API slightly smoother.

Reply:
I think this mixes the logic for creating a call info and the logic for resolving the calls. For instance, when resolving method calls, we defer until call resolution to compute the scope of the composite type on which the method is being called. It also breaks precedent, since `CallInfo`s don't contain the call and POI scopes -- so calls to `resolveCall` would deliver one scope (in a hidden way) through the `CallInfo`, and two explicitly using arguments? Computing the ID as an out-parameter also matches the `actualAsts` out-parameter that `CallInfo::create` already had.

Reply:
OK, works for me. Could you clarify in a comment here describing `moduleScopeId` to include what ID it is giving back (is it the ID of `Submodule` in `Module.Submodule.fn()` or is it the ID of `Submodule` itself?)